lib/collection/src/collection_state.rs

```diff
@@ -34,12 +37,34 @@ impl State {
             channel_service,
         )
         .await?;
+        Ok(())
+    }
+
+    async fn apply_shard_transfers(
+        shard_transfers: HashSet<ShardTransfer>,
+        collection: &Collection,
+        this_peer_id: PeerId,
+        mut abort_transfer: impl FnMut(ShardTransfer),
+    ) -> CollectionResult<()> {
+        let old_transfers = collection
+            .shards_holder
+            .read()
+            .await
+            .shard_transfers
+            .clone();
+        for transfer in shard_transfers.difference(&old_transfers) {
+            if transfer.from == this_peer_id {
+                // Abort transfer as sender should not learn about the transfer from snapshot
+                // If this happens it mean the sender is probably outdated and it is safer to abort
+                abort_transfer(transfer.clone())
```

Comment:
Is this case worth a log line?

Reply:
We have a `log::warn` at the place where we execute abort transfer, there `reason` will also be reported.